arch/arm/core/Kconfig

```diff
@@ -31,6 +31,22 @@ config CPU_HAS_SYSTICK
 	help
 	  This option is enabled when the CPU has systick timer implemented.
 
+config CPU_HAS_ARM_MPU
```

Comment:
I don't really like this idea, here, i.e. adding SOC-specific Kconfig options to reflect the flavors of the mPU. 
This file should only have arm/core/ - specific symbol definitions, IMO

Reply:
Could we bring back the ARM_CORE_MPU option, just as an intermediate abstraction-option to be used for _all_ possible MPU flavors?

The clean-up of the SOC/Board-specific MPU support options will _still_ be relevant, of course.

Reply:
I not catching your idea.

Proposed scheme looks like this:
- CPU_HAS_XXX_MPU signals what MPU is present on  given platform.
- ARM_MPU enables MPU support. There is no other option controlling that.
- If ARM_MPU is enabled, the directory with MPU drivers is included in the compilation. Common files are compiled in unconditionally, while MPU drivers are compiled in only if relevant CPU_HAS_XXX_MPU option is present (there will be only one at a time).

Where you see ARM_CORE_MPU in this scheme?

IMHO removal of CPU_HAS_XXX_MPU options, which are defined by SoC, destroys the whole ideas behind this PR:
- Create single option controlling MPU support.
- Automatically select correct driver on each platform.

Reply:
@pizi-nordic all this looks ok. I just do not like so much the idea of having CPU_HAS_xxx options in arch/arm/core/Kconfig

Reply:
This file should _ideally_ be used to define arm-core specific symbols, not CPU capabilities. But otherwise, the clean-up is good, and I a like it.

Reply:
I wonder if these defines could be moved to the parent Kconfig file i.e. arch/arm/Kconfig.

Reply:
No problem :)

Reply:
I think so, yes. I would prefer the arch/arm/core/Kconfig to include ARM features, not SOC-specific (CPU_HAS_) capabilities.

Perhaps @galak or @MaureenHelm could comment here, too?

Reply:
Then,  I believe, we came to a point where we would need to introduce a soc/arm/Kconfig file, to host all symbols that signify soc-specific capabilities/configuration for ARM MCUs.

Would you find this useful, @MaureenHelm? IMO, this would be a step towards keeping arch/arm as "SOC-agnostic" as possible.

@pizi-nordic I wonder if you could make a try of this.

Reply:
Moved to `arch/arm/Kconfig`.